src/settings.js

```diff
@@ -1363,10 +1363,25 @@ var EMIT_EMSCRIPTEN_METADATA = 0;
 // for effective code size optimizations to take place.
 var SUPPORT_ERRNO = 1;
 
-// Internal: points to a file that lists all asm.js/wasm module exports, annotated
-// with suppressions for Closure compiler, that can be passed as an --externs file
-// to Closure.
+// Internal: An array of all symbols exported from asm.js/wasm module.
 var MODULE_EXPORTS = [];
 
 // Internal (testing only): Disables the blitOffscreenFramebuffer VAO path.
 var OFFSCREEN_FRAMEBUFFER_FORBID_VAO_PATH = 0;
+
+// If true, uses minimal sized runtime without POSIX features, Module, preRun/preInit/etc.,
+// Emscripten built-in XHR loading or library_browser.js. Enable this setting to target
+// the smallest code size possible.
+// Set MINIMAL_RUNTIME=2 to further enable even more code size optimizations. These opts are
+// quite hacky, and work around limitations in Closure and other parts of the build system, so
+// they may not work in all generated programs (But can be useful for really small programs)
+var MINIMAL_RUNTIME = 0;
+
+// If building with MINIMAL_RUNTIME=1 and application uses sbrk()/malloc(), enable this. If you
+// are not using dynamic allocations, can set this to 0 to save code size. This setting is
+// ignored when building with -s MINIMAL_RUNTIME=0.
+var USES_DYNAMIC_ALLOC = 1;
+
+// Specifies the set of runtime JS functions that should be imported to the asm.js/wasm module.
+// Remove elements from this list to make build smaller if some of these are not needed.
+var RUNTIME_FUNCS_TO_IMPORT = ['abort', 'assert', 'setTempRet0', 'getTempRet0']
```

Comment:
metadce removes such things automatically, so can mention that this is not needed for wasm in higher optimization levels (-O3/-Os and above)